numpy/core/src/multiarray/ctors.c

```diff
@@ -422,6 +422,10 @@ copy_and_swap(void *dst, void *src, int itemsize, npy_intp numitems,
     }
 }
 
+PyObject *
+_array_from_array_like(PyObject *op, PyArray_Descr *requested_dtype,
+                       npy_bool writeable, PyObject *context);
+
```

Comment:
The function should be marked `static` or `NPY_NO_EXPORT`

Reply:
I think static is correct for file-local functions

Reply:
`git grep 'static PyObject'` certainly returns a lot of results

Reply:
Done.